src/gan/discriminator/model.py

```diff
@@ -1,72 +1,213 @@
+                          out_channels=self.num_filters, 
+                          kernel_size=sz, 
+                          stride=1, 
+                          padding=0),
+                nn.ReLU(),
+                nn.MaxPool1d(kernel_size=2),
+                nn.Flatten()) for sz in self.filter_sizes
+            ]
+        )
+
+        self.out = nn.Sequential(
+                        nn.Dropout(self.dropout_prob2),
+                        nn.Linear(self.channels_out, self.hidden_dims),
+                        nn.Linear(self.hidden_dims, self.num_classes)
+                      )
+        
+        for block in self.conv_blocks:
+            block.apply(self.init_weights)
+
+        self.out.apply(self.init_weights)
+
+    def init_weights(self, m):
+        if type(m) in (nn.Linear, nn.Conv1d):
+            nn.init.kaiming_uniform_(m.weight)
+            
+            if m.bias is not None:
+                nn.init.constant_(m.bias, 0)
+
+    def forward(self, verbatim_indices):
+        embeddings = self.emb(verbatim_indices)
+        embeddings = embeddings.permute(0, 2, 1)
+        X = self.dropout1(embeddings)
+
+        X = [conv(embeddings) for conv in self.conv_blocks]
+        X_cat = torch.cat(X, 1)
+
+        return self.out(X_cat)
+
+    def predict(self, X):
+        self.eval()
+
+        with torch.no_grad():
+            yhat = self(X).softmax(dim=-1)
+
+            scores, labels = yhat.topk(1, -1, True, True)
+            return scores, labels
+
+    def train_single_code(self, train):
+        criterion = nn.CrossEntropyLoss()
+        #acc_metric = accuracy
+
+        return self.trainer(train, criterion, 
+                     self.args['early_stopping'])
+
+
+    def trainer(self, train, criterion, early_stopping=True,
+                n_epochs_stop=10):
+
+        trainloader = DataLoader(train, batch_size=self.args['batch_size'], shuffle=True, num_workers=4)
+        steps = len(trainloader)
+
+        #acc = [0] * steps
+        self.optimizer = torch.optim.Adam(filter(lambda p: p.requires_grad, 
+                                                 self.parameters()), 
+                                                 #lr=0.0001,
+                                                 amsgrad=True)
+        #epochs_no_improve = 0
+        #max_test_acc = 0
+        if self.args['device'] == 'cuda':
+            self.cuda()
+            self.optimizer.cuda()
+
+        self.train()
+
+        for i, (x, y) in enumerate(tqdm(trainloader)):
+            verbs = x.to(self.args['device'])
+            labels = y.to(self.args['device'])
+        
+            # Forward pass
+            outputs = self.forward(verbs)
+            loss = criterion(outputs, labels.argmax(1))
+            
+            # Backward and optimize
+            self.optimizer.zero_grad()
+            loss.backward()
+            self.optimizer.step()
+
+        return loss.item()
+
+            # self.eval()
```

Comment:
This is extremely minor, but if this whole block is being commented out, can it be removed? (I'm generally not a fan of including commented-out code. It's a little confusing.)